pkg/config/config.go

```diff
@@ -26,11 +28,17 @@ import (
 )
 
 type Config interface {
```

Comment:
nit: Add comments on public (exposed) structs and functions.

The same for elsewhere.

Reply:
I will have other PRs for comments